src/llm/abstract_llm.py

```diff
@@ -0,0 +1,80 @@
+from typing import List, Dict
+
+from langchain.chains.llm import LLMChain
+from langchain.memory.chat_memory import BaseChatMemory
+from langchain.prompts import ChatPromptTemplate
+
+class AbstractLLM:
+    def __init__(self, config, session_id=None):
+        """
+        :param config: Configuration dictionary
+
+        The constructor of the AbstractLLM class allows users to pass
+        a configuration dictionary to the LLM. This configuration dictionary
+        can be used to configure the LLM temperature, prompt and other
+        necessities.
+        """
+        if config is None or not isinstance(config, dict):
```

Comment:
We could simplify this verification

```suggestion
        if not isinstance(config, dict):
```